tests/tests_geomstats/test_product_manifold.py

```diff
@@ -9,277 +11,286 @@
-        normalized_logs = gs.einsum(
-            "..., ...j->...j",
-            1.0 / self.space_vector.metric.norm(logs, base_point),
-            logs,
-        )
-        point = self.space_vector.metric.exp(normalized_logs, base_point)
-        result = self.space_vector.metric.dist(point, base_point)
-
-        expected = gs.ones(n_samples)
-        self.assertAllClose(result, expected)
-
-    @geomstats.tests.np_autograd_and_torch_only
-    def test_dist_log_exp_norm_matrix(self):
+    def test_dist_log_exp_norm(self, manifold, n_samples):
         n_samples = 10
-        point = self.space_matrix.random_point(n_samples)
-        base_point = self.space_matrix.random_point(n_samples)
-        logs = self.space_matrix.metric.log(point, base_point)
+        point = manifold.random_point(n_samples)
+        base_point = manifold.random_point(n_samples)
+        logs = manifold.metric.log(point, base_point)
         normalized_logs = gs.einsum(
             "..., ...jl->...jl",
-            1.0 / self.space_matrix.metric.norm(logs, base_point),
+            1.0 / manifold.metric.norm(logs, base_point),
             logs,
         )
-        point = self.space_matrix.metric.exp(normalized_logs, base_point)
-        result = self.space_matrix.metric.dist(point, base_point)
+        point = manifold.metric.exp(normalized_logs, base_point)
+        result = manifold.metric.dist(point, base_point)
         expected = gs.ones((n_samples,))
         self.assertAllClose(result, expected)
 
     @geomstats.tests.np_autograd_and_torch_only
-    def test_inner_product_matrix_matrix(self):
-        euclidean = Euclidean(3)
-        minkowski = Minkowski(3)
-        space = ProductManifold(
-            manifolds=[euclidean, minkowski], default_point_type="matrix"
-        )
-        point = space.random_point(1)
-        result = space.metric.metric_matrix(point)
-        expected = gs.eye(6)
-        expected[3, 3] = -1
-        self.assertAllClose(result, expected)
-
-    @geomstats.tests.np_autograd_and_torch_only
-    def test_inner_product_matrix_vector(self):
-        euclidean = Euclidean(3)
-        minkowski = Minkowski(3)
-        space = ProductManifold(
-            manifolds=[euclidean, minkowski], default_point_type="vector"
-        )
-        point = space.random_point(1)
-        expected = gs.eye(6)
-        expected[3, 3] = -1
-        result = space.metric.metric_matrix(point)
-        self.assertAllClose(result, expected)
-
-    def test_regularize_vector(self):
-        expected = self.space_vector.random_point(5)
-        result = self.space_vector.regularize(expected)
-        self.assertAllClose(result, expected)
-
-    def test_regularize_matrix(self):
-        expected = self.space_matrix.random_point(5)
-        result = self.space_matrix.regularize(expected)
-        self.assertAllClose(result, expected)
-
-    @geomstats.tests.np_autograd_and_torch_only
-    def test_inner_product_matrix(self):
```

Comment:
Where did this go? maybe we can rename it `test_log_inner_product_squared_dist`?